std/range/package.d

```diff
@@ -2442,7 +2442,8 @@ have at most one element.
 
 The type returned by $(D takeOne) is a random-access range with length
 regardless of $(D R)'s capabilities (another feature that distinguishes
-$(D takeOne) from $(D take)).
+$(D takeOne) from $(D take)). One exception: The range can save() only
+if $(D R) can.
```

Comment:
So I thought a bit about this and technically the status quo for `takeOne` is:

> If only provided with an `InputRange`, `takeOne` is neither a `RandomAccess` nor a `ForwardRange`

_but_ the spec also says:

> `r.front` evaluated multiple times, without calling `r.popFront`, or otherwise mutating the range object or the underlying data, yields the same result for every evaluation.

So before your PR it would have been possible to provide `save` even for a `InputRange`, because a call to `front` is always guaranteed to yield the same element and the range can be "saved" as currently by storing it's emptiness, the source range and not calling `popFront` on the source range.

The spec only has been updated to this tight definition [quite recently](https://github.com/dlang/phobos/pull/4511) (since 2.072), before it was allowed to yield different results for multiple calls on `front` and e.g. most famously [generate](https://github.com/dlang/phobos/pull/4528/files) did so.

CC @JackStouffer and @schveiguy (the authors of this change)